modules/core/src/main/scala/fs2/kafka/KafkaConsumer.scala

```diff
@@ -178,13 +179,34 @@ object KafkaConsumer {
                 )
               )
 
-            def fetchPartition(deferred: Deferred[F, PartitionRequest]): F[Unit] = {
-              val request = Request.Fetch(
-                partition,
-                streamId,
-                deferred.complete(_: PartitionRequest).void
-              )
-              val fetch = requests.offer(request) >> deferred.get
+            def fetchPartition: F[Unit] = F.deferred[PartitionResult].flatMap { deferred =>
+              val callback: PartitionResult => F[Unit] =
+                deferred.complete(_).void
+
+              val fetch: F[PartitionResult] = withPermit {
+                val assigned =
+                  withConsumer.blocking {
+                    _.assignment.contains(partition)
+                  }
+
+                def storeFetch: F[Unit] =
+                  actor.ref.modify { state =>
```

Comment:
Looks a bit weird that we need to modify internal actor state from the another component.

Reply:
Yeah, I guess it's an artifact of an iterative transition away from having the actor be a separate encapsulated component. Maybe we should reconceptualise the state as "consumer state" rather than "actor state".

Reply:
Up to you. I mentioned it only to make sure that this is intentional.